lib/ClangImporter/ClangImporter.cpp

```diff
@@ -1709,12 +1709,16 @@ bool ClangImporter::isModuleImported(const clang::Module *M) {
   return M->NameVisibility == clang::Module::NameVisibilityKind::AllVisible;
 }
 
-bool ClangImporter::canImportModule(ImportPath::Element moduleID) {
+bool ClangImporter::canImportModule(ImportPath::Module path) {
+  if (path.hasSubmodule()) {
+    // TODO: more efficient way to check for submodule availability?
+    return bool(loadModule(path.front().Loc, path));
```

Comment:
Is loading the module an acceptable way to check? What would be a better approach?

I tried to iterate through `submodules()` of `clang::Module` directly, but it traps at finding the iterator front. I assume you have to load the module to find its submodules?

Reply:
Unfortunately, this approach is too expensive and potentially source-breaking. `canImport` checks are frequently run by clients that expect lightning-fast responses from parse-only passes. We need to use the appropriate lookup paths instead. Additionally, loading modules potentially makes those symbols visible.

Reply:
Make sense. I'll try to figure out how to do that.